docs/learning/workshop_a_precision_converter_fpga_integration_journey/index.rst

```diff
@@ -0,0 +1,693 @@
+
+*4 AXI-Stream interfaces*
+- CMD: Instructions
+- SDO: SPI write data (MOSI)
+- SDI: SPI read data (MISO)
+- SYNC: Synchronization events
+
+*Each streams follows standard AXI-Stream handshaking*
+- Only the main AXI-stream signals are used (ready, valid, data)
+
+**SPI Engine Framework – Software support**
+
+- Introduces the concept of SPI offload
+- Moves certain operations typically performed by the application processor to the SPI controller
+- Interrupt offload capability
+- Data offload capability
+- SPI-Engine is one implementation of SPI offload
+- ADI converter drivers can be used with any offload capable SPI controller
+
+
+**Build prerequisites – HDL repository**
+
+.. figure:: build_prerequisites_hdl.png
+   :align: center
+
+**Build prerequisites – Linux repository**
+
+.. figure:: build_prerequisites_linux.png
+   :align: center
+
+.. figure:: build_prerequisites_clone.png
+   :align: center
+
+Use Case
+~~~~~~~~
+
+**What is the use case?**
+
+.. figure:: use_case_examples.png
+   :align: center
+
+*Requirements:*
+- Highest sampling rate possible with low jitter in sample acquisition
+- Maximum SNR
+- Minimum THD
+- Minimize processor usage for data acquisition and processing
+
+.. figure:: use_case_typical_app.png
+   :align: center
+
+.. figure:: use_case_snr.png
+   :align: center
+
+.. figure:: use_case_jitter.png
+   :align: center
+
+============================ ====================== =====================
+Test conditions              Regular SPI controller SPI Engine controller
+============================ ====================== =====================
+Resolution[bits]             16                     18
+Sampling rate[KSPS]          15                     15 and 1330
+Input signal frequency[kHz]  1                      1
+Input signal amplitude[dBFS] -0.5                   -0.5
+Supply voltage [V]           -2.5 and 5             -2.5 and 5
+============================ ====================== =====================
+
+**Features of the AD7984 converter**
+
+- High performance
+- Throughput: 1.33 MSPS
+- True differential analog input range: ±VREF
+- 0 V to VREF with VREF between 2.9 V to 5 V
+- Zero latency architecture
+- 18-bit resolution with no missing codes
+- Dynamic range: 99.7 dB, VREF = 5 V
+- SNR: 98.5 dB at fIN = 1 kHz, VREF = 5 V
+- THD: −110.5 dB at fIN = 1 kHz, VREF = 5 V
+- SINAD: 97.5 dB at fIN = 1 kHz, VREF = 5 V
+
+**AD7984 SPI transfer Timing Diagram**
+
+.. figure:: use_case_ad7984_transfer_diagram.png
+   :align: center
+
+**Timing parameters needed to configure the framework**
+
+.. figure:: use_case_timing_parameters_ad7984.png
+   :align: center
+
+**HDL design block diagram**
+
+.. figure:: use_case_hdl_bd.png
+   :align: center
+
+**HDL Framework instantiation**
+
+- TCL Function header:
+
+proc spi_engine_create {{name "spi_engine"} {data_width 32} {async_spi_clk 1} {num_cs 1} {num_sdi 1} {sdi_delay 0} {echo_sclk 0}}
+
+- Instantiation example for PulSAR ADC:
+
+  .. code:: tcl
```

Comment:
```suggestion
  .. code-block:: tcl
```
All "code" directives should be replaced with code-block

Reply:
fixed